### ISSUE TITLE
Subscriber Role Shows Nav Links The Don't Have Permissions to View

### ISSUE THREAD

**Issue by ash1eygrace** [Automattic/jetpack#23975]
### Impacted plugin

Jetpack

### Steps to Reproduce

1. Invite a WordPress.com User as a Subscriber on an Atomic or Jetpack connected site via My Site > Users invite. 
2. Log in to the Subscriber's WordPress.com account > Click on My Site > Choose the site you're subscribed to. 
3. Click on the links you shouldn't have access to as a subscriber and get various permission on-page errors, banners, or Console errors.

### A clear and concise description of what you expected to happen.

I'd expect to see a link to view the site's front end since I'm only a subscriber. 

At the least, I'd expect not to see the links that my role can't access, as they just load broken pages due to lack of permission. 

### What actually happened

I see the following links in the dashboard of the site I'm subscribed to:

- Stats
- Upgrades
- Posts
- Media
- Pages
- Comments
- Feedback
- Jetpack
- Appearance
- Plugins
- Users
- Tools
- Settings
- WP Admin

When I click on any of the Nav links above other than published Pages or Posts, I get permission errors, which is expected since subscribers can only view posts, and pages, and leave comments if enabled. Technically, the role-based permissions are working (I can't access things I don't have permission to as a subscriber), but it looks really broken and alarming when every page either won't load or tells me I don't have permission to view the page: 

https://user-images.githubusercontent.com/29527450/163636153-b75c791a-3e21-4694-a03b-b3860ba6f67b.mp4

 

### Operating System

macOS

### Browser

Chrome/Chromium

**Comment by coder-karen** [comment]
Yes I can also see the same - only if inviting a user as a subscriber though via the WordPress.com interface with the steps you gave (but not adding them as a subscriber to the site itself in wp-admin). An example of an error I see is that it appears I can activate new themes, but in attempting to do so I see a warning message saying 'Unable to activate theme. Contact support.'. It would make sense to hide some of these menu items.

**Comment by jeherve** [comment]
Nice catch!

Since that interface (the WordPress.com dashboard) is developed in the Calypso repo, I'll transfer the issue there.
